packages/nx/src/command-line/exec/exec.ts

```diff
@@ -117,26 +165,43 @@ function ensureNxTarget(project: ProjectGraphProjectNode, targetName: string) {
   }
 }
 
-function getProject(projectGraph: ProjectGraph) {
-  const projectName = calculateDefaultProjectName(
+function getDefaultProject(
+  projectGraph: ProjectGraph
+): ProjectGraphProjectNode | undefined {
+  const defaultProjectName = calculateDefaultProjectName(
     process.cwd(),
     workspaceRoot,
     readProjectsConfigurationFromProjectGraph(projectGraph),
     readNxJson()
   );
+  if (defaultProjectName && projectGraph.nodes[defaultProjectName]) {
+    return projectGraph.nodes[defaultProjectName];
+  }
+}
 
-  if (!projectName) {
-    output.error({
-      title: 'Unable to determine project name for `nx exec`',
-      bodyLines: [
-        "`nx exec` should be ran from within an Nx project's root directory.",
-        'Does this package.json belong to an Nx project?',
-      ],
-    });
-    process.exit(1);
+function getProjects(
+  projectGraph: ProjectGraph,
+  nxArgs: NxArgs
+): ProjectGraphProjectNode[] {
+  let selectedProjects = {};
+
+  // get projects matched
+  if (nxArgs.projects?.length) {
+    for (const projectName of nxArgs.projects) {
+      selectedProjects[projectName] = projectGraph.nodes[projectName];
+    }
+  } else {
+    // if no project specified, return all projects
+    for (const projectName in projectGraph.nodes) {
+      selectedProjects[projectName] = projectGraph.nodes[projectName];
+    }
```

Comment:
nit: could be more precise
```suggestion
      selectedProjects = ...projectGraph.nodes;
```

Reply:
changed